test/test_cuda.py

```diff
@@ -3060,8 +3060,12 @@ def test_graph_optims(self):
         ] + [
             (optimizer_ctor, {"lr": 0.1, "betas": (0.8, 0.7), "fused": True, "amsgrad": amsgrad})
             for optimizer_ctor, amsgrad in product((torch.optim.Adam, torch.optim.AdamW), (False, True))
+        ] + [
+            (torch.optim.ASGD, {"lr": 0.1, "foreach": True, "maximize": maximize, "weight_decay": weight_decay})
```

Comment:
Is single tensor not capturable today?

If this PR only adds capturable for foreach, please add an error message in the constructor and also in the single-tensor impl to let people know that capturable on single tensor is not supported. Also include a note in the docs.